pkg/controllers/management/globaldns/globaldns_handler.go

```diff
@@ -0,0 +1,228 @@
+package globaldns
+
+import (
+	"context"
+	"net"
+	"sort"
+
+	v1beta1Rancher "github.com/rancher/types/apis/extensions/v1beta1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	clientv1beta1 "k8s.io/client-go/kubernetes/typed/extensions/v1beta1"
+)
+
+const (
+	GlobaldnsController = "mgmt-global-dns-controller"
+)
+
+type GlobalDNSController struct {
+	globalDNSInterface v3.GlobalDNSInterface
```

Comment:
nit: we have some sort of best practices in naming listers and clients. For lister, it is `<object>Lister`, for clients - it is `<object>s`

Reply:
done